rpc/client/client.go

```diff
@@ -692,31 +695,60 @@ func (c *Client) BlockSearch(ctx context.Context, query string, page, perPage *i
 func (c *Client) Status(ctx context.Context) (*ctypes.ResultStatus, error) {
 	latest, err := c.node.Store.LoadBlock(c.node.Store.Height())
 	if err != nil {
-		// TODO(tzdybal): extract error
 		return nil, fmt.Errorf("failed to find latest block: %w", err)
 	}
 
-	latestBlockHash := latest.Header.DataHash
-	latestAppHash := latest.Header.AppHash
-	latestHeight := latest.Header.Height
-	latestBlockTimeNano := latest.Header.Time
+	initial, err := c.node.Store.LoadBlock(uint64(c.node.GetGenesis().InitialHeight))
+	if err != nil {
+		return nil, fmt.Errorf("failed to find earliest block: %w", err)
+	}
+
+	validators, err := c.node.Store.LoadValidators(latest.Header.Height)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch the validator info at latest block: %w", err)
+	}
+	_, validator := validators.GetByAddress(latest.Header.ProposerAddress)
+
+	state, err := c.node.Store.LoadState()
+	if err != nil {
+		return nil, fmt.Errorf("failed to load the last saved state: %w", err)
+	}
+	defaultProtocolVersion := corep2p.NewProtocolVersion(
+		version.P2PProtocol,
+		state.Version.Consensus.Block,
+		state.Version.Consensus.App,
+	)
+	id, addr, network := c.node.P2P.Info()
+	txIndexerStatus := "on"
 
 	result := &ctypes.ResultStatus{
-		// TODO(tzdybal): NodeInfo
+		NodeInfo: corep2p.DefaultNodeInfo{
+			ProtocolVersion: defaultProtocolVersion,
+			DefaultNodeID:   id,
+			ListenAddr:      addr,
+			Network:         network,
+			Version:         rconfig.Version,
+			Moniker:         config.DefaultBaseConfig().Moniker,
+			Other: corep2p.DefaultNodeInfoOther{
+				TxIndex:    txIndexerStatus,
+				RPCAddress: c.config.ListenAddress,
+			},
+		},
 		SyncInfo: ctypes.SyncInfo{
-			LatestBlockHash:   latestBlockHash[:],
-			LatestAppHash:     latestAppHash[:],
-			LatestBlockHeight: int64(latestHeight),
-			LatestBlockTime:   time.Unix(0, int64(latestBlockTimeNano)),
-			// TODO(tzdybal): add missing fields
-			//EarliestBlockHash:   earliestBlockHash,
-			//EarliestAppHash:     earliestAppHash,
-			//EarliestBlockHeight: earliestBlockHeight,
-			//EarliestBlockTime:   time.Unix(0, earliestBlockTimeNano),
-			//CatchingUp:          env.ConsensusReactor.WaitSync(),
+			LatestBlockHash:     latest.Header.DataHash[:],
+			LatestAppHash:       latest.Header.AppHash[:],
+			LatestBlockHeight:   int64(latest.Header.Height),
+			LatestBlockTime:     time.Unix(0, int64(latest.Header.Time)),
+			EarliestBlockHash:   initial.Header.DataHash[:],
+			EarliestAppHash:     initial.Header.AppHash[:],
+			EarliestBlockHeight: int64(initial.Header.Height),
+			EarliestBlockTime:   time.Unix(0, int64(initial.Header.Time)),
+			CatchingUp:          true, // the client is always syncing in the background to the latest height
```

Comment:
[non-blocking]  Can this field ever be false, today or in the future? Returning `CatchingUp` in status suggests that we are actively checking if the node is syncing. In theory it could not be syncing but we return true.

Reply:
I feel the first condition is a valid one. The second one is not as we are deleting the map at the end of `TrySyncBlock` and could end up with an empty map just after syncing a block even though there are blocks left to sync.

Reply:
if there is a chance of not receiving the header (hence no `syncTarget` trigger) and but received the block from DA layer (leading to non-empty `syncCache`)? If so, it is better to check `syncCache` as a fallback.

Reply:
Good Point could happen. We should also consider what we mean by `CatchingUp`. As beeing in the `SyncLoop` could already be considered as `CatchingUp`. Therefore we might want to test for the opposite. As in, is it already catched up ? So we check if `syncTarget = m.store.Height() && syncCache empty ` => false else true.

Reply:
We should double check what `CatchingUp` means in tendermint, To be sure we return "compatible" result. 
My intuition is that if "syncTarget` > `m.store.Height()+1`, we;re catching up, otherwise we're "synced" (not catching up anymore).

Reply:
Created follow-up issue: https://github.com/celestiaorg/rollmint/issues/625